portal-api/api-tests/v1/users/post-create-user.api-test.js

```diff
@@ -0,0 +1,238 @@
+        const newUser = {
+          ...MOCK_USER,
+          password: 'No-numeric-characters',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.password).toEqual(PASSWORD_ERROR);
+      });
+
+      it('rejects if the provided password contains zero upper-case characters', async () => {
+        const newUser = {
+          ...MOCK_USER,
+          password: '0-upper-case-characters',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.password).toEqual(PASSWORD_ERROR);
+      });
+
+      it('rejects if the provided password contains zero lower-case characters', async () => {
+        const newUser = {
+          ...MOCK_USER,
+          password: '0-LOWER-CASE-CHARACTERS',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.password).toEqual(PASSWORD_ERROR);
+      });
+
+      it('rejects if the provided password contains zero special characters', async () => {
+        const newUser = {
+          ...MOCK_USER,
+          password: '0specialCharacters',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.password).toEqual(PASSWORD_ERROR);
+      });
+
+      it('rejects if the provided password contains fewer than 8 characters', async () => {
+        const newUser = {
+          ...MOCK_USER,
+          password: '1234567',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.password).toEqual(PASSWORD_ERROR);
+      });
+    });
+
+    describe('validating the email address', () => {
+      it('rejects if the provided email address is not in valid format', async () => {
+        const newUser = {
+          ...MOCK_USER,
+          email: 'abc',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.email.text).toEqual(EMAIL_ERROR.text);
+      });
+
+      it('rejects if the provided email address is empty', async () => {
+        const newUser = {
+          ...MOCK_USER,
+          email: '',
+        };
+
+        const { status, body } = await createUser(newUser);
+
+        expect(status).toEqual(400);
+        expect(body.success).toEqual(false);
+        expect(body.errors.errorList.email.text).toEqual(EMAIL_ERROR.text);
+      });
```

Comment:
For simplicity -- as this is never going to be visible on the front end (as we only have an `email` field in `portal`), we only have two rules -- one checking the `email` is an email address, and the other checking the `username` matches. There is no need to check the `username` is an email address -- as it cannot get through otherwise, and the only way someone would get these errors would be if they posted directly to `portal-api` -- in which case we assume a familiarity of `portal-api`. Basically - YAGNI